scripts/resolve_suggestions_urls.py

```diff
@@ -0,0 +1,58 @@
+import json
+from jsonpointer import resolve_pointer, set_pointer
+
+VERSION = "v4.0.0"
+
+
+def find_pointers_containing(input_data, search_key, pointer=None):
+    if isinstance(input_data, dict):
+        if pointer and search_key in input_data:
+            yield pointer
+        for k, v in input_data.items():
+            yield from find_pointers_containing(
+                v, search_key, f"{pointer}/{k}" if pointer else f"/{k}"
+            )
+    elif isinstance(input_data, list):
+        for index, item in enumerate(input_data):
+            yield from find_pointers_containing(item, search_key, f"{pointer}/{index}")
+
+
+def resolve_schema(schema_address, version=None, language=None):
+    with open(schema_address, "r+") as file:
+        data = json.load(file)
+        pointer_iterator = find_pointers_containing(data, "suggestions_url")
+        for pointer in pointer_iterator:
+            address = resolve_pointer(data, "{}{}".format(pointer, "/suggestions_url"))
+            if language:
+                set_pointer(
+                    data,
+                    "{}{}".format(pointer, "/suggestions_url"),
+                    address.format(version=version, language_code=language),
+                )
+            else:
+                set_pointer(data, "{}{}".format(pointer, "/suggestions_url"), "")
```

Comment:
Seeing this made me realise I should have suggested using the `find_pointers_to` rather than `find_pointers_containing` method. The downside to this would be more copying of methods from the translations repo. On looking into this however I noticed that there is another option. We could use the `jsonpath_rw` library to search for the elements. Something like:

```suggestion
        json_path = parse("$..suggestions_url")
        for match in json_path.find(data):
            json_pointer = json_path_to_json_pointer(str(match.full_path))
            suggestions_url = match.value
            if language:
                set_pointer(
                    data,
                    json_pointer,
                    suggestions_url.format(version=version, language_code=language),
                )
            else:
                set_pointer(data, json_pointer, "")
```

Then we could remove the `find_pointers_containing` method and replace it with a copy of `json_path_to_json_pointer`, which is a much smaller method.

Reply:
Refactored and working fine, thanks!👍